### PR TITLE
Extract model names, to obtain a human-readable name for each model

### PR DESCRIPTION
**Pull request by ruiAzevedo19**
Part of #206

### CODE REVIEW

**Comment by ruiAzevedo19** [comment]
Based on https://github.com/symflower/eval-dev-quality/pull/216

File: model/llm/llm.go [hunk 1]
```diff
@@ -123,6 +125,16 @@ func (m *Model) ID() (id string) {
...
 
+// Name returns the name of this model.
+func (m *Model) Name() (name string) {
+	return m.name
+}
+
+// SetName sets the name of this model.
+func (m *Model) SetName(name string) {
+	m.name = name
+}
```
[the remark this answers is not in the document]
Comment (bauersimon) [reply model/llm/llm.go:136]: I think you will essentially have two constructors. One bare-minimum for the ID (that also the openai API provider can use). And then another one with all the additional information.

**Review by bauersimon — CHANGES_REQUESTED** [review]

File: model/llm/llm.go [hunk 2]
```diff
@@ -123,6 +125,16 @@ func (m *Model) ID() (id string) {
 	return m.model
 }
 
+// Name returns the name of this model.
+func (m *Model) Name() (name string) {
+	return m.name
+}
```
Comment (bauersimon) [inline model/llm/llm.go:131]: TODO this also needs to go into the CSV

**Comment by bauersimon** [comment]
Please try this out with the cheapest model from openrouter and post the CSV here to see how it looks like.

**Comment by ruiAzevedo19** [comment]
@bauersimon Results

### OpenRouter

**Command**
```
eval-dev-quality evaluate --runs 1 --repository golang/plain --model openrouter/meta-llama/llama-3-8b-instruct
```

<details><summary>evaluation.csv</summary>

```
model-name,model-id,cost,language,repository,task,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
Meta: Llama 3 8B Instruct,openrouter/meta-llama/llama-3-8b-instruct,0.00000014,golang,golang/plain,write-tests,3,0,0,171,2114,181,1,1,1
```

</details>

<details><summary>golang-summed.csv</summary>

```
model-name,model-id,cost,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
Meta: Llama 3 8B Instruct,openrouter/meta-llama/llama-3-8b-instruct,0.00000014,3,0,0,171,2114,181,1,1,1
```

</details>

<details><summary>models-summed.csv</summary>

```
model-name,model-id,cost,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
Meta: Llama 3 8B Instruct,openrouter/meta-llama/llama-3-8b-instruct,0.00000014,3,0,0,171,2114,181,1,1,1
```

</details>

### Ollama

**Command**
```
eval-dev-quality evaluate --runs 1 --repository golang/plain --model ollama/qwen:0.5b
```

<details><summary>evaluation.csv</summary>

```
model-name,model-id,cost,language,repository,task,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
qwen:0.5b,ollama/qwen:0.5b,0,golang,golang/plain,write-tests,2,0,0,155,8592,1041,1,0,1

```

</details>

<details><summary>golang-summed.csv</summary>

```
model-name,model-id,cost,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
qwen:0.5b,ollama/qwen:0.5b,0,2,0,0,155,8592,1041,1,0,1
```

</details>

<details><summary>models-summed.csv</summary>

```
model-name,model-id,cost,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
qwen:0.5b,ollama/qwen:0.5b,0,2,0,0,155,8592,1041,1,0,1
```

</details>

### OpenAi

**Command**
```
eval-dev-quality evaluate --runs 1 --repository golang/plain --urls custom-ollama:http://127.0.0.1:11434/v1 --model custom-ollama/qwen:0.5b

```

<details><summary>evaluation.csv</summary>

```
model-name,model-id,cost,language,repository,task,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
qwen:0.5b,custom-ollama/qwen:0.5b,0,golang,golang/plain,write-tests,2,0,0,159,4697,600,1,0,1
```

</details>

<details><summary>golang-summed.csv</summary>

```
model-name,model-id,cost,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
qwen:0.5b,custom-ollama/qwen:0.5b,0,2,0,0,159,4697,600,1,0,1
```

</details>

<details><summary>models-summed.csv</summary>

```
model-name,model-id,cost,score,coverage,files-executed,generate-tests-for-file-character-count,processing-time,response-character-count,response-no-error,response-no-excess,response-with-code
qwen:0.5b,custom-ollama/qwen:0.5b,0,2,0,0,159,4697,600,1,0,1
```

</details>

**Comment by bauersimon** [comment]
Please still always put the model id first.

**Comment by bauersimon** [comment]
Just asking myself if we should put the model names in quotes in the csv... As long as no name **contains** a `,` it should be good.

**Comment by bauersimon** [comment]
> Just asking myself if we should put the model names in quotes in the csv... As long as no name **contains** a `,` it should be good.

Not necessary: https://pkg.go.dev/encoding/csv#Writer.Write.
Will quote as required.

File: cmd/eval-dev-quality/cmd/evaluate.go [hunk 3]
```diff
@@ -196,7 +196,8 @@ func (command *Evaluate) Initialize(args []string) (evaluationContext *evaluate.
 				command.logger.Panicf("ERROR: unknown custom provider %q for model %q", providerID, model)
 			}
 
-			modelProvider.AddModel(llm.NewModel(modelProvider, model))
+			modelName := strings.Split(model, provider.ProviderModelSeparator)
+			modelProvider.AddModel(llm.NewNamedModelWithCost(modelProvider, model, modelName[len(modelName)-1], 0))
```
Comment (bauersimon) [inline cmd/eval-dev-quality/cmd/evaluate.go:200]: Can use https://pkg.go.dev/strings#SplitN to get just the model name.

`strings.Split(model, provider.ProviderModelSeparator, 1)[1]`

**Review by bauersimon — CHANGES_REQUESTED** [review]

**Review by bauersimon — APPROVED** [review]

File: cmd/eval-dev-quality/cmd/evaluate.go (hunk 3 above)
Comment (bauersimon) [reply cmd/eval-dev-quality/cmd/evaluate.go:200]: Wait why was this not resolved... I guess it should've been `strings.SplitN(model, provider.ProviderModelSeparator, 2)[1]`.

**Pull request merged**